image-qemu.c

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (c) 2018 Alexandre Fournier <alexandre.fournier@kiplink.fr>, Kiplink
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <confuse.h>
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+#include <errno.h>
+
+#include "genimage.h"
+
+struct qemu {
+	const char *format;
+};
+
+static int qemu_generate(struct image *image)
+{
+	struct partition *part;
+	struct qemu *qemu = image->handler_priv;
+	char *partitions = NULL;
+	int ret;
+
+	list_for_each_entry(part, &image->partitions, list) {
+		struct image *child;
+		const char *infile;
+
+		image_log(image, 1, "adding partition '%s'%s%s%s ...\n", part->name,
+			part->image ? " from '": "",
+			part->image ? part->image : "",
+			part->image ? "'" : "");
+
+		child = image_get(part->image);
+		infile = imageoutfile(child);
+
+		if (!partitions)
+			xasprintf(&partitions, "%s", infile);
+		else
+			xasprintf(&partitions, "%s %s", partitions, infile);
```

Comment:
We're working on properly quoting all file names in a different PR. Please quote all paths with single quotes here.